### PR TITLE
docs: clarify amounts in microstacks

### PR DESCRIPTION
**Pull request by obycode**
### Description
There is no mention of microstacks in this documentation, which could be
confusing for new developers. This commit specifies the units for the
amounts used in  `stx-get-balance`, `stx-account`, `stx-transfer?` and
`stx-burn?`.

I did not see any consistency with line-wrapping, so I just eye-balled it to something that looked reasonable compared to the existing strings, but would be happy to make changes to that if you want.

One thing I am unclear on is whether https://docs.stacks.co/docs/write-smart-contracts/clarity-language/language-functions gets generated from this file or if I should also open a PR one that repo.

### Applicable issues
N/A

### Checklist
N/A - documentation improvement only
- [ ] Test coverage for new or modified code paths
- [ ] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [ ] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

File: clarity/src/vm/docs/mod.rs [hunk 1]
```diff
@@ -1912,8 +1912,8 @@ const STX_GET_BALANCE: SimpleFunctionAPI = SimpleFunctionAPI {
     signature: "(stx-get-balance owner)",
     description: "`stx-get-balance` is used to query the STX balance of the `owner` principal.
 
-This function returns the STX balance of the `owner` principal. In the event that the `owner`
-principal isn't materialized, it returns 0.
+This function returns the STX balance, in microstacks (1 STX = 1000000 microstacks), of the
```
Comment (jcnelson) [inline clarity/src/vm/docs/mod.rs:1915]: Can you write 1000000 as 1,000,000 so it's clear that it's 1 million?  Thanks!

File: clarity/src/vm/docs/mod.rs [hunk 2]
```diff
@@ -1978,8 +1979,9 @@ This function returns (ok true) if the transfer is successful, or, on an error,
 const STX_BURN: SimpleFunctionAPI = SimpleFunctionAPI {
     name: None,
     signature: "(stx-burn? amount sender)",
-    description: "`stx-burn?` decreases the `sender` principal's STX holdings by `amount`, destroying
-the STX. The `sender` principal _must_ be equal to the current context's `tx-sender`.
+    description: "`stx-burn?` decreases the `sender` principal's STX holdings by `amount`,
+specified in microstacks, destroying the STX. The `sender` principal _must_ be equal to the current
```
Comment (jcnelson) [inline clarity/src/vm/docs/mod.rs:1983]: `destroying the STX` --> `by destroying them`

**Review by jcnelson — APPROVED** [review]
LGTM. Just small nits

File: clarity/src/vm/docs/mod.rs (hunk 2 above)
Comment (obycode) [reply clarity/src/vm/docs/mod.rs]: Addressed in 2bb0ee182.

File: clarity/src/vm/docs/mod.rs (hunk 1 above)
Comment (obycode) [reply clarity/src/vm/docs/mod.rs]: Addressed in 2bb0ee182.

File: clarity/src/vm/docs/mod.rs [hunk 3]
```diff
@@ -1912,8 +1912,8 @@ const STX_GET_BALANCE: SimpleFunctionAPI = SimpleFunctionAPI {
     signature: "(stx-get-balance owner)",
     description: "`stx-get-balance` is used to query the STX balance of the `owner` principal.
 
-This function returns the STX balance of the `owner` principal. In the event that the `owner`
```
Comment (pavitthrap) [inline clarity/src/vm/docs/mod.rs:1915]: I think it could be appropriate to specify `1 STX = 1 million microstacks` under the top heading of `Functions`.

**Review by pavitthrap — APPROVED** [review]
lgtm, but one minor suggestion

File: clarity/src/vm/docs/mod.rs (hunk 3 above)
Comment (obycode) [reply clarity/src/vm/docs/mod.rs:1915]: Thanks Pavi. I think that would have to go into the [docs repo](https://github.com/stacks-network/docs), right?

**Pull request merged**